### DISCUSSION TITLE
JS file update issue on Supabase Storage

### DISCUSSION THREAD

**Discussion by NOTIFYJ** [supabase/supabase#12124] [Questions]
I am encountering a small issue with the storage on Supabase. I uploaded a JS file to my storage and attempted to update it while keeping the same sharing link to avoid any inconvenience for users. However, I have noticed that users who installed the script before the update do not have access to the new version, while those who installed it after the update do have access to the updated version.

My question is: is it possible to update the JS file for all users of this file?

Thank you for your help.

Best regards,

NB : the method I used to make the update I did is to delete old file containing js from supabase and upload the new version with same name.js

**Comment by GaryAustin1** [comment]
Files will be cached in the browser and the CDN until the original cache time expires. 

See this  https://supabase.com/docs/guides/storage/cdn#cache-duration

**Comment by NOTIFYJ** [comment]
I am aware that caching can have a limited duration, but how can I ensure
immediate updates for users accessing the JavaScript code through this
link? Many users are not finding updates, or even all of them.
